lib/Sema/CSApply.cpp

```diff
@@ -6708,6 +6708,23 @@ Expr *ExprRewriter::coerceToType(Expr *expr, Type toType,
 
       maybeDiagnoseUnsupportedFunctionConversion(cs, expr, toFunc);
 
+      if (auto *closure = dyn_cast<ClosureExpr>(expr)) {
```

Comment:
This looks similar to existing logic in ExprRewriter - can you consolidate?

Reply:
Sure, going to try, this is more of an experiment stage right now.